models/project/board.go

```diff
@@ -307,6 +307,11 @@ func SetDefaultBoard(ctx context.Context, projectID, boardID int64) error {
 		return err
 	}
 
+	_, err = GetBoard(ctx, boardID)
+	if err != nil {
+		return err
+	}
```

Comment:
That call should be before `#302`, otherwise we have a project without default board again.

Reply:
Yup, that's what I mentioned before: there might be various cases that there is no default board (either by existing bug, or newly introduced bug in the future). So self-healing is good enough.

---

Or ideally they could be in a transaction, then no worry about inconsistency.

Reply:
Moved